src/core/credentials/transport/tls/grpc_tls_certificate_provider.cc

```diff
@@ -89,9 +93,14 @@ absl::Status ValidatePemKeyCertPair(absl::string_view cert_chain,
   for (X509* x509 : *parsed_certs) {
     X509_free(x509);
   }
+
+  if (!std::holds_alternative<std::string>(private_key)) {
```

Comment:
Instead of checking which type the variant contains twice (once here, and again on line 103), let's just do it once:

```
std::string* private_key = std::get_if<std::string>(&private_key);
if (private_key == nullptr) return absl::OkStatus();
// Check that the private key consists of valid PEM blocks.
absl::StatusOr<EVP_PKEY*> parsed_private_key =
    ParsePemPrivateKey(*private_key);
```